resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -372,24 +390,22 @@ private[spark] class Client(
       .map(CompressionUtils.createTarGzip(_))
   }
 
-  private def getDriverLauncherService(
-      k8ClientConfig: Config,
-      kubernetesMaster: String): KubernetesSparkRestApi = {
-    val url = s"${
-      Array[String](
-        kubernetesMaster,
-        "api", "v1", "proxy",
-        "namespaces", namespace,
-        "services", kubernetesAppId).mkString("/")}" +
-      s":$DRIVER_LAUNCHER_SERVICE_PORT_NAME/"
-
-    val sslContext = SSLUtils.sslContext(k8ClientConfig)
-    val trustManager = SSLUtils.trustManagers(
-      k8ClientConfig)(0).asInstanceOf[X509TrustManager]
-    HttpClientUtil.createClient[KubernetesSparkRestApi](
-      uri = url,
-      sslSocketFactory = sslContext.getSocketFactory,
-      trustContext = trustManager)
+  private def getDriverLauncherClient(
+      kubernetesClient: KubernetesClient,
+      service: Service): KubernetesSparkRestApi = {
+    val servicePort = service
+      .getSpec
+      .getPorts
+      .asScala
+      .filter(_.getName == DRIVER_LAUNCHER_SERVICE_PORT_NAME)
+      .head
+      .getNodePort
+    // NodePort is exposed on every node, so just pick one of them.
```

Comment:
The port should be exposed on all nodes and we can pick any at random. Using the Master's port seems less than ideal because we're trying to shift traffic away from it by not using the apiserver proxy.